packages/router/router.ts

```diff
@@ -2627,22 +2628,19 @@ async function callLoaderOrAction(
         "Redirects returned/thrown from loaders/actions must have a Location header"
       );
 
-      // Check if this an external redirect that goes to a new origin
-      let currentUrl = new URL(request.url);
-      let currentOrigin = currentUrl.origin;
-      let newOrigin = new URL(location, currentOrigin).origin;
```

Comment:
This was the problem - since the origin for a `test://test.com` URL used in Remix integration tests is `null` 😕 , and `new URL(path, "null")` throws an error.

<img width="612" alt="Screenshot 2022-12-06 at 9 32 36 AM" src="https://user-images.githubusercontent.com/1609022/205939912-ce830d0e-95c6-4c13-b41d-86f85d407c46.png">